src/app/ui/NewTaskForm.tsx

```diff
@@ -107,7 +107,7 @@ export const NewTaskForm = ({ handleCreate, handleClose, getSignedUrlUpload }: N
   useEffect(() => {
     function handleEscPress(e: KeyboardEvent) {
       if (e.key === 'Escape') {
-        handleClose()
+        handleClose({ triggeredFrom: 'MOUSE' })
```

Comment:
Instead of adding `triggeredFrom: 'Mouse'` to every button and icon, we can make it the default behavior

Reply:
Using a default value for `triggeredFrom` can lead to implicit behavior, making it unclear what triggered the modal close if `triggeredFrom` is omitted by mistake. Explicitly specifying 'MOUSE' or 'KEYBOARD' keeps the intent clear and improves readability. This approach avoids potential debugging issues due to unintended default behavior.

Reply:
@aatbip I think we only care when trigged  by keyboard, isn't it? In that case, we could make it simpler by making this optional parameter. Which should've kept the changes minimal. But now that you have made the changes, I do agree with you here that it keeps intention clear. But the way it is done, looks bothersome to provide the triggeredFrom params everywhere.

Reply:
Changes added in the commit below @rrojan @priosshrsth